### DISCUSSION TITLE
New feature - A "simple" Measurement ruler

### DISCUSSION THREAD

**Discussion by cryoglobe** [supermerill/SuperSlicer#1429] [Community contribution]
Quite often . I would like to have a ruler from inside the slicer .. so I can easily and quicly take a measure between two points
What I have to do know is create a box shape with the size I wants and move it between the two points I want to check , and scale it to see the size .. ( not very accurate )

That ruler should be available from the 3D view , to check a scaling effect ... 
The outside size of the object is not always the one we want to check .... :)

But it could also be nice from the sliced view / gcode preview .. to be able to verify the effect of some parameters on the final quotes ..

**Comment by supermerill** [comment]
That's a big gui job.
Also it's very hard to pin a point in the 3D space with a mouse. it's easier with a VR wand & VR headset...
Often, you have to resort to a 2-step clic: first select the xy position, by projecting the mouse onto the bed, and then select the height you want. But even with that it's hard to get the right point in a flat 2D screen.

As the gui stuff was made (& most of the time modified, upgraded) by prusa, i would recommend opening an issue on their github to politely ask them to add this ruler to their awesome slicer gui (like you did here).

**Reply by neophyl** [reply]
This has been a feature request over at the Prusa github for a quite awhile. https://github.com/prusa3d/PrusaSlicer/issues/3651

**Comment by cryoglobe** [comment]
Hello

Thanks for the answer

Indeed , it's exactly what I was looking for ... and better described
I'll submit the idea as you suggested it

Thanks again

Kind regards

**Reply by supermerill** [reply]
like neophyl said, just answer back to prusa3d#3651 to reinforce the need.
